src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -115,34 +145,86 @@ internal async Task<IEnumerable<IProjectVersionedValue<RestoreData>>> RestoreAsy
             };
         }
 
-        private async Task<bool> RestoreCoreAsync(ProjectRestoreInfo restoreInfo)
+        private async Task<bool> RestoreCoreAsync(PackageRestoreUnconfiguredInput value)
         {
-            // Restore service always does work regardless of whether the value we pass 
-            // them to actually contains changes, only nominate if there are any.
-            byte[] hash = RestoreHasher.CalculateHash(restoreInfo);
+            ProjectRestoreInfo? restoreInfo = value.RestoreInfo;
+            bool success = false;
 
-            if (_latestHash != null && Enumerable.SequenceEqual(hash, _latestHash))
-                return true;
+            try
+            {
+                // Prevent extra restore under some conditions.
+                if (IsProjectConfigurationVersionOutOfDate(value.ConfiguredInputs))
+                {
+                    return false;
+                }
 
-            _latestHash = hash;
+                // Restore service always does work regardless of whether the value we pass 
+                // them to actually contains changes, only nominate if there are any.
+                byte[] hash = RestoreHasher.CalculateHash(restoreInfo!);
```

Comment:
Rather than `!` here, consider using `Assumes.NotNull(restoreInfo)`.

Reply:
Change done in this `https://github.com/dotnet/project-system/pull/7199/commits/adcd21bf069711920a47adbfd6ee78e18d52aaab`